include/rta/supernode.h

```diff
@@ -15,9 +15,11 @@ namespace cryptonote {
     class transaction;
 }
 
+namespace graft::supernode::request { struct SupernodeAnnounce; }
+
 namespace graft {
 
-struct SupernodeAnnounce;
+using graft::supernode::request::SupernodeAnnounce;
```

Comment:
If you do this then you can see graft::SupernodeAnnounce everywhere. Does it make sense?

Reply:
everywhere inside of this compile unit. Is it problem?

Reply:
Inside every compile unit which includes this header file. In general - avoid "using" in public headers.

Reply:
I'm ready to use fully specified names in signatures.
That is
Now is (my version and that guys want to FIX):
bool updateFromAnnounce(const SupernodeAnnounce &announce);
Guy's proper version:
bool updateFromAnnounce(const graft::supernode::request::SupernodeAnnounce &announce);

Which one is better?
Boris?
My way - easier to read. And it will save time for everyone who will read it many many times.
It's the only reason for me.

Reply:
btw
SupernodeAnnounce - could be just Announce
Because it's all inside of supernode. And there are not any other announces.